### DISCUSSION TITLE
Max retries exceeded with url

### DISCUSSION THREAD

**Discussion by iamguruman** [eternnoir/pyTelegramBotAPI#1522] [General]
Hello

Bot worked fine few weeks. But today in the morning Ive get an error:

requests.exceptions.ConnectionError: HTTPSConnectionPool(host='api.telegram.org', port=443): Max retries exceeded with url: /bot................/getUpdates?offset=1&timeout=20 (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f5170ac72e0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))

after couple hours while I tried to solve the problem... I restart bot and it work fine... Im not fix anything in the code, only restart

after couple minutes Ive get again this error

In the internet Ive found solutions with proxy, my opinion proxy will give not long term solution, Im not clear understand why I need to use proxy to solve the problem. Please describe me why proxy is a solution? Another solutions?

**Comment by coder2020official** [comment]
Hello. This error happens due to internet problems.

Possible reasons:
- bad internet
- api.telegram.org blocked by your provider
Proxy may fix your issue only if api.telegram.org is blocked by your provider(but ig it is not blocked because your bot is working for some time though)

So, you are having connection issues due to bad internet.

**Reply by iamguruman** [reply]
ok, thx
